hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -752,16 +770,85 @@ public Hashtable<String, String> getBlobMetadata(final Path path,
+        // Check here if the path is an implicit dir
+        if (ex.getStatusCode() == HttpURLConnection.HTTP_NOT_FOUND) {
+          List<BlobProperty> blobProperties = getListBlobs(
+              path,null, null, tracingContext, 2, true);
+          if (blobProperties.size() == 0) {
+            throw ex;
+          }
+          else {
+            // The path was an implicit blob. Create Marker and set metadata on it
+            createDirectory(path, null, FsPermission.getDirDefault(),
+                FsPermission.getUMask(
+                    getAbfsConfiguration().getRawConfiguration()),
+                tracingContext);
+
+            boolean xAttrExists = metadata.containsKey(HDI_ISFOLDER);
+            if (!xAttrExists) {
+              metadata.put(HDI_ISFOLDER, TRUE);
+              metadataRequestHeaders = getRequestHeadersForMetadata(metadata);
+            }
+
+            final AbfsRestOperation op = client.setBlobMetadata(path, metadataRequestHeaders, tracingContext);
+            perfInfo.registerResult(op.getResult()).registerSuccess(true);
+          }
+        }
+        else {
+          throw ex;
+        }
+      }
+    }
+  }
+
+  /**
+   * Gets user-defined properties(metadata) of the container over blob endpoint.
+   * @param tracingContext
+   * @return hashmap containing key value pairs for container metadata
+   * @throws AzureBlobFileSystemException
+   */
+  public Hashtable<String, String> getContainerMetadata(TracingContext tracingContext)
+      throws AzureBlobFileSystemException {
+    try (AbfsPerfInfo perfInfo = startTracking("getContainerMetadata", "getContainerMetadata")) {
+      LOG.debug("getContainerMetadata for filesystem: {}", client.getFileSystem());
+
+      final AbfsRestOperation op = client.getContainerMetadata(tracingContext);
+      perfInfo.registerResult(op.getResult()).registerSuccess(true);
+
+      final Hashtable<String, String> metadata = parseResponseHeadersToHashTable(op.getResult());
+      return metadata;
+    }
+  }
+
+  /**
+   * Sets user-defined properties(metadata) of the container over blob endpoint.
+   * @param metadata set of user-defined properties to be set
+   * @param tracingContext
+   * @throws AzureBlobFileSystemException
+   */
+  public void setContainerMetadata(final Hashtable<String, String> metadata,
+      TracingContext tracingContext) throws AzureBlobFileSystemException {
+    try (AbfsPerfInfo perfInfo = startTracking("setContainerMetadata", "setContainerMetadata")) {
+      LOG.debug("setContainerMetadata for filesystem: {} with properties: {}",
+          client.getFileSystem(),
+          metadata);
+
       final List<AbfsHttpHeader> metadataRequestHeaders = getRequestHeadersForMetadata(metadata);
-      final AbfsRestOperation op = client.setBlobMetadata(
-          path, metadataRequestHeaders, tracingContext);
+      final AbfsRestOperation op = client.setContainerMetadata(metadataRequestHeaders, tracingContext);
```

Comment:
common code with getFileSystemProperties.

Reply:
You mean setFileSystemProperties right??
Only two lines of code are getting repeated and both of them are function calls themselves...
I think it would be redundant to have another method that we will still need to call from two places.

What do you think??
We will only be saving one line repetition.